rust/agama-migrate-wicked/src/reader.rs

```diff
@@ -20,9 +20,9 @@ fn replace_colons(colon_string: String) -> String {
     replaced
 }
 
-pub async fn read_dir(path: PathBuf) -> Result<Vec<Interface>, io::Error> {
-    let interfaces = fs::read_dir(path)?
+pub async fn read_dir(path: PathBuf) -> Result<Vec<Vec<Interface>>, io::Error> {
```

Comment:
I think it would be more consistent to keep it like:
```suggestion
pub async fn read_dir(path: PathBuf) -> Result<Vec<Interface>, io::Error> {
```
and use something like [flatten()](https://doc.rust-lang.org/std/iter/trait.Iterator.html#examples-25)